src/pages/index.js

```diff
@@ -0,0 +1,284 @@
+  if (evt.target.classList.contains("modal")) {
+    closeModal(evt.target);
+  }
+}
+
+function closeModalEsc(evt) {
+  if (evt.key === "Escape") {
+    const modal = document.querySelector(".modal_opened");
+    closeModal(modal);
+  }
+}
+
+function handleLike(event, id, isLiked) {
+  api
+    .changeLike(id, isLiked)
+    .then(() => {
+      event.target.classList.toggle("card__like-btn_liked");
+    })
+    .catch(console.error);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  buttonText(submitBtn, true, "Delete", "Deleting...");
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      buttonText(submitBtn, false, "Delete", "Deleting");
+    });
+}
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  buttonText(submitBtn, true);
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+      disableButton(cardSubmitBtn, config);
+    })
+    .catch(console.error)
+    .finally(() => {
+      buttonText(submitBtn, false, "Saving...", "Save");
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  buttonText(submitBtn, true, "Saving...", "Save");
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  api
+    .addCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      disableButton(cardSubmitBtn, config);
+      cardNameInput.value = "";
+      cardLinkInput.value = "";
+      evt.target.reset();
+      closeModal(cardModal);
+    })
+    .catch(console.error);
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  buttonText(submitBtn, true, "Saving...", "Save");
+  avatarSubmitBtn.disabled = true;
+  api
+    .editAvatarInfo(avatarLinkInput.value)
+    .then((data) => {
+      userAvatar.src = data.avatar;
+      closeModal(avatarModal);
+      avatarForm.reset();
+      disableButton(avatarSubmitBtn, config);
+    })
+    .catch(console.error)
+    .finally(() => {
+      buttonText(submitBtn, false);
+    });
+}
+
+avatarModalBtn.addEventListener("click", () => {
+  avatarLinkInput.value = "";
+  resetValidation(avatarForm, [avatarLinkInput], {
+    inactiveBtnSelector: "modal__submit-btn_type_disabled",
+    errorClass: "modal__input_type_error",
+  });
+  openModal(avatarModal);
+});
+
+avatarModalCloseBtn.addEventListener("click", () => {
```

Comment:
The `avatar` close icon listener will not be needed anymore because it will be handled universally in line 263 with the other close icons